Pix4D Rest API v3 Upload Custom Template Demo.py

```diff
@@ -0,0 +1,163 @@
+    "client_id": client_id,
+    "client_secret": client_secret,
+    "token_format": "jwt",
+  }
+  query_params = "&".join([f"{k}={v}" for k, v in query_params_dict.items()])
+  url = f"{CLOUD_URL}/oauth2/token/?{query_params}"
+  resp = requests.post(url)
+  resp.raise_for_status()
+  return resp.json()["access_token"]
+
+
+def headers(token):
+  return {"Authorization": f"Bearer {token}"}
+
+
+def create_project(name, token):
+  url = f"{PROJECT_URL}/"
+  resp = requests.post(url, json={"name": name}, headers=headers(token))
+  resp.raise_for_status()
+  return resp.json()
+
+
+def project_s3_creds(project_id, token, share_token=None):  # share_token?
+  url = f"{PROJECT_URL}/{project_id}/s3_credentials/"
+  resp = requests.get(url, headers=headers(token), params=_get_params(share_token))
+  resp.raise_for_status()
+  return resp.json() 
+
+
+def _get_params(share_token=None):
+  params = {}
+  if share_token is not None:
+    params["shareToken"] = share_token
+  return params
+
+
+def project_s3_creds(project_id, token, share_token=None):
+  url = f"{PROJECT_URL}/{project_id}/s3_credentials/"
+  resp = requests.get(url, headers=headers(token), params=_get_params(share_token))
+  resp.raise_for_status()
+  return resp.json() 
+
+
+def register_images(project_id, token, image_keys):
+  url = f"{PROJECT_URL}/{project_id}/inputs/bulk_register/"
+  resp = requests.post(
+    url, headers=headers(token), json={"input_file_keys": image_keys}
+  )
+  resp.raise_for_status()
+  return resp.json
+
+
+def register_extra_file(project_id, token, file_key):
+  url = f"{PROJECT_URL}/{project_id}/extras/"
+  resp = requests.post(
+    url, headers=headers(token), json={"file_key": file_key}
+  )
+  resp.raise_for_status()
+  return resp.json
+
+
+def start_processing(project_id, token, processing_options=None):
+  url = f"{PROJECT_URL}/{project_id}/start_processing/"
+  resp = requests.post(url, headers=headers(token), data=processing_options)
+  resp.raise_for_status()
+  return resp.json
+
+
+# main -------------
+def main():
+  print("main started")
+
+  # Get access token
+  my_jwt = get_jwt(PIX4D_CLIENT_ID, PIX4D_CLIENT_SECRET)
+  
+  # Create project with a random project id
+  project_id = create_project(f"demo {randint(0, 1000)}", my_jwt)['id']
+  
+  # Get AWS S3 credentials
+  s3_creds = project_s3_creds(project_id, my_jwt)
+  
+  # Create service client at s3.
+  s3_client = boto3.client("s3",
+                            aws_access_key_id=s3_creds["access_key"],
+                            aws_secret_access_key=s3_creds["secret_key"],
+                            aws_session_token=s3_creds["session_token"])
+  
+  
+  # Uploading imanges. "image" directory is supposed to be in the same directory as this script.
+  keys = []
+  images = list(Path("images/").glob("*.JPG"))
+
+  with ProgressBar(max_value=len(images)) as pbar:
+    for i, image in enumerate(images):
+      k = str(Path(s3_creds["key"]) / Path(image).name)
```

Comment:
```suggestion
      k = str(Path(s3_creds["key"]) / "images" / Path(image).name)
```

Makes S3 more readable if the images are put in a subdirectory

Reply:
I will modify accordingly.